### PR TITLE
Fix memory leak in OpenSSL deinitialization function

### PR DESCRIPTION
**Pull request by macvk**
Changes proposed in this pull request:
 - Fixed memory leaks in OpenSSL deinitialization function.

### CODE REVIEW

**Comment by chipitsine** [comment]
Travis is failing

**Comment by macvk** [comment]
> 
> 
> Travis is failing

I don't think so

**Comment by macvk** [comment]
I don't see any error in appveyor
Could someone clarify the appveyor error?

**Comment by davidebeatrici** [comment]
`Maximum allowed artifact storage size of 90000 Mb will be exceeded.`, not related to the pull request.

Could you clean your `master` branch or create a dedicated one and a new pull request, please?

File: src/Mayaqua/Encrypt.c [hunk 1]
```diff
@@ -3684,6 +3685,10 @@ void FreeOpenSSLThreadState()
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
 	CRYPTO_cleanup_all_ex_data();
 	ERR_remove_thread_state(NULL);
+#else
+#ifndef LIBRESSL_VERSION_NUMBER
```
Comment (davidebeatrici) [inline src/Mayaqua/Encrypt.c]: The two lines can be combined into one: `#elif !defined(LIBRESSL_VERSION_NUMBER)`

Comment (macvk) [reply src/Mayaqua/Encrypt.c]: Thanks. I'd like to keep my code unchanged.

**Comment by macvk** [comment]
> Could you clean your `master` branch or create a dedicated one and a new pull request, please?

sorry, what should I clean? I've changed only 3 files, 3 lines of code in total. Please clarify what should I do? if I create new branch, it'll contain the same 3 files + 3 lines of code.

if the issue is in appveyor, maybe you should disable it temporary?

**Comment by macvk** [comment]
I've noticed that almost all latest pull requests has the same error... even pull requests which was accepted by you guys.

**Comment by davidebeatrici** [comment]
> sorry, what should I clean? I've changed only 3 files, 3 lines of code in total. Please clarify what should I do? if I create new branch, it'll contain the same 3 files + 3 lines of code.

Look at the commits.

**Comment by macvk** [comment]
It is a bug in appveyor. I've found a lot of same issues. As for my commit it changes only 3 lines of code. I think any next commit to this project will fail with the same appveyor error.

**Comment by chipitsine** [comment]
please, do not pay attention to app veyor, we are out of build artifacts quota.

can you squash commits into one ?

**Comment by davidebeatrici** [comment]
> It is a bug in appveyor. I've found a lot of same issues. As for my commit it changes only 3 lines of code. I think any next commit to this project will fail with the same appveyor error.

The AppVeyor build failing is not the issue. There are 4 unrelated commits in this pull request.

**Comment by chipitsine** [comment]
well, we pay some attention to app veyor builds (probably, we should remove artifacts).
anyway, we are slowly going to Azure Pipelines (I am a bit busy last month)

**Comment by macvk** [comment]
all these commits was made by me and already was merged into SE by the owner, see here for instance https://github.com/SoftEtherVPN/SoftEtherVPN/pull/202/commits/1ba1afb1c0b95beda5cadc13652f74af61fcf62e

so what should I do? I can't fork twice one project. I need create new account on github just to do this pool request ?  or rebase my fork? what happened with my prev commits if I erase them now?

**Comment by davidebeatrici** [comment]
Ideally, you should rebase your `master` branch against the one in this repository and then create a new branch from it each time you want to create a pull request.

**Comment by macvk** [comment]
I don't think it is possible to do everything ideally. If I rebase my fork, the prev commits will be lost.

**Comment by davidebeatrici** [comment]
Squash your last 3 commits and cherry-pick them in another branch.
You can then cherry-pick those commits in your `master` branch again if you don't want to create a new pull request.

**Comment by macvk** [comment]
I've just tried to make a new branch and make new pull request from it. but this pull request contains the same list of commits.

**Comment by davidebeatrici** [comment]
You have to pull forcibly from this repository's `master` branch.

**Comment by macvk** [comment]
what the problem to apply pull request in this state? only 3 file changes.

**Comment by davidebeatrici** [comment]
The extra commits would probably end up in our `master` branch.

**Comment by macvk** [comment]
the master branch doesn't contain commits from 2015 till 2018. so it's probably not ideal at the moment.

**Comment by chipitsine** [comment]
@macvk  , do you mind if we will provide you short guide how to squash commits ?

**Comment by davidebeatrici** [comment]
I fixed everything for you.

**Comment by macvk** [comment]
thank you.

**Comment by davidebeatrici** [comment]
You're welcome.

**Comment by davidebeatrici** [comment]
Thank you very much for your contribution!

**Pull request merged**